### ISSUE TITLE
test_gzip_load error with Python 3.5

### ISSUE THREAD

**Issue by d10genes** [numpy/numpy#6329]
I'm trying out python 3.5, and wasn't sure numpy installed properly, so I ran the tests and got this error:

    $ python -c "import numpy; numpy.test()"
    ...

    ERROR: test_io.test_gzip_load
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/me/py-env/p3/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
        self.test(*self.arg)
      File "/Users/me/py-env/p3/lib/python3.5/site-packages/numpy/lib/tests/test_io.py", line 1675, in test_gzip_load
        assert_array_equal(np.load(f), a)
      File "/Users/me/py-env/p3/lib/python3.5/site-packages/numpy/lib/npyio.py", line 381, in load
        fid.seek(-N, 1)  # back-up
      File "/Users/me/py-env/p3/lib/python3.5/site-packages/numpy/lib/npyio.py", line 52, in seek
        offset = self.offset + offset
    AttributeError: 'GzipFile' object has no attribute 'offset'

    ----------------------------------------------------------------------
    Ran 5556 tests in 29.421s

    FAILED (KNOWNFAIL=6, SKIP=18, errors=1)

I wasn't sure if the tests are expected to run without failing or not on Python 3.5, thought I would report it just in case.

I have the following installed with this virtual environment:

```
gnureadline==6.3.3
ipython==3.2.1
Jinja2==2.8
jsonschema==2.5.1
MarkupSafe==0.23
mistune==0.7.1
nose==1.3.7
numpy==1.9.2
ptyprocess==0.5
Pygments==2.0.2
pyzmq==14.7.0
scipy==0.16.0
terminado==0.5
tornado==4.2.1
wheel==0.24.0
```

**Comment by njsmith** [comment]
Known problem, will be fixed by 1.10 (which is due out imminently). It
should only affect using numpy io functions to work with gzip files, so it
may not affect you in any case.

**Comment by matthew-brett** [comment]
I am hoping to get a patched version of 1.9.2 out ASAP to fix this - see : https://github.com/numpy/numpy/issues/6319

**Comment by matthew-brett** [comment]
In the mean time I do have a patched 1.9.2 OSX wheel up here:

```
pip wheel -f https://nipy.bic.berkeley.edu/scipy_installers numpy==1.9.2
```

**Comment by njsmith** [comment]
Going to close this since fixes are ongoing in other bugs.
